includes/Media/Video/Muting.php

```diff
@@ -201,6 +200,10 @@ public function update_callback_is_muted( $value, $object ) {
 			);
 		}
 
-		return (bool) update_metadata( $meta_type, $object_id, $meta_key, (int) $value );
+		$value = rest_sanitize_boolean( $value );
+
+		update_post_meta( $object_id, $meta_key, $value );
+
+		return true;
```

Comment:
```suggestion
		return (bool) update_post_meta( $object_id, $meta_key, $value );;
```

I think this is cleaner.

Reply:
Attention. This has different semantics.

I changed this on purpose. Specifically to fix the actual bug at play here.

For two reasons:

1. The return value doesn't matter and is not used if it's not a `WP_Error`
2. `update_post_meta` returns the meta ID if the key didn't exist, `true` on successful update, `false` on failure or if the value passed to the function is the same as the one that is already in the database.

So "returns false if the value is already the same as the one that is already in the database". That is really dangerous!

It means if `true` is in the database and we save `true` again, we get `false`. That's not what we want!

Thus, this does not work. Which is why I changed it.

Reply:
Okay, thanks for the context, it is useful.